### PR TITLE
Add mechanism for arbitrary head tags

### PR DESCRIPTION
**Pull request by marcoow**
This adds a mechanism for defining arbitrary head tags. It introduces a `HeadTags` class that is injected as `headTags` into all components and that has 2 methods:

* `write` writes a head tag or updates its attributes or content if it exists already; it takes a `tagName`, `keys` (attributes that are used to check if the tag exists already), `values` (attributes that are not used to check if the tag exists already) and `textContent`
* `remove` removes a tag from the `<head>` again; it takes it takes a `tagName` and `keys` (attributes that are used to check if the tag exists already)

This is the foundation for #654. This MR also adds the following tags:

* `meta[property="article:published_time"]` for blog posts
* `meta[property="fb:admins"]`
* `meta[property="og:site_name"]`
* `meta[name="twitter:site"]`

### CODE REVIEW

File: src/utils/head-tags.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+export default class HeadTags {
+
+
+  public static create(): HeadTags {
+    return new HeadTags();
+  }
+  private document: HTMLDocument = window.document;
+
+  public write(tagName, keyAttrs = {}, contentAttrs = {}, textContent = null) {
```
Comment (pichfl) [inline src/utils/head-tags.ts:9]: I'm not sure why `keyAttrs` and `contentAttrs` are split here if they end up getting merged. Even for the component usage, I think having a single object makes for a better dev experience.

**Review by pichfl — CHANGES_REQUESTED** [review]
Very elegant

File: src/utils/head-tags.ts (hunk 1 above)
Comment (marcoow) [reply src/utils/head-tags.ts:9]: The key (😄) here is that `keyAttrs` are

> attributes that are used to check if the tag exists already

while `contentAttrs` are not. If a tag `meta[name="twitter:site"]` exists with a different `content` attribute we want to update it vs. adding another one. `HeadTags` needs to be able to understand which attributes should be used to find an element that should potentially be updated vs. which attributes to ignore for looking up the element.

Comment (pichfl) [reply src/utils/head-tags.ts:9]: I think documenting this to some extend might be helpful. I see how it works in the code after your explanation.

**Review by pichfl — APPROVED** [review]

**Pull request merged**